cmd/spire-agent/cli/run/run.go

```diff
@@ -389,11 +389,43 @@ func parseTrustBundle(bundleBytes []byte, trustBundleContentType string) ([]*x50
 	return nil, fmt.Errorf("unknown trust bundle format: %s", trustBundleContentType)
 }
 
-func downloadTrustBundle(trustBundleURL string) ([]byte, error) {
+func downloadTrustBundle(trustBundleURL string, serverAddress string, serverPort int, trustDomain string) ([]byte, error) {
+	var req *http.Request
+	u, err := url.Parse(trustBundleURL)
+	if err != nil {
+		return nil, err
+	}
+	client := &http.Client{}
+	if u.Scheme == "unix" {
+		socketPath := u.Path
+		client = &http.Client{
+			Transport: &http.Transport{
+				DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+					return net.Dial("unix", socketPath)
+				},
+			},
+		}
+		params := u.Query()
+		params.Set("trust-domain", trustDomain)
+		params.Set("server-address", serverAddress)
+		params.Set("server-port", strconv.Itoa(serverPort))
+		params.Set("mode", "bootstrap")
+		u.RawQuery = params.Encode()
+		u.Scheme = "http"
+		u.Host = "localhost"
+		u.Path = "/"
+		req, err = http.NewRequest("GET", u.String(), nil)
+	} else {
+		req, err = http.NewRequest("GET", trustBundleURL, nil)
```

Comment:
In my view, we should always perform a GET request to the configured trustBundleURL. The difference would be the in the transport used by the HTTP client.

Reply:
the trust bundle url wont be a valid http url though? I dont think http.NewRequest would be happy to see a 'unix:///some/path/to/unix.sock'?

Reply:
I just double checked this, It does fail if you just pass trustBundleURL

Exception:
```
unsupported protocol scheme "unix"
```

We have to rewrite the url that it sees so it functions properly.